Samples~/ClientSimIssue3RespawnTest/ClientSimIssue3RespawnTest.cs

```diff
@@ -0,0 +1,170 @@
+            Assert.IsNotNull(testHelpers, "Could not find Test helper reference.");
+            
+            // Listen for OnPlayerRespawn event through dispatcher
+            bool respawnEventFromDispatcherTriggered = false;
+            void OnPlayerRespawn(ClientSimOnPlayerRespawnEvent respawnEvent)
+            {
+                respawnEventFromDispatcherTriggered = true;
+            }
+            EventDispatcher.Subscribe<ClientSimOnPlayerRespawnEvent>(OnPlayerRespawn);
+
+            // Close the menu before doing anything.
+            Helper.CloseMenu();
+            
+            // Walk into the respawn cube.
+            Helper.TestInput.SetInputRun(true);
+
+            yield return Helper.WalkToPoint(testHelpers.respawnWithIndexCube.gameObject.transform, "Didn't reach RespawnWithIndexCube");
+
+            // Wait for the same amount of time the UdonBehaviour uses to delay the Respawn call
+            yield return new WaitForSeconds(testHelpers.respawnWithIndexCube.GetProgramVariable<float>(_programVariableDelay));
+
+            // Verify that the player is very close to Spawn
+            Vector3 playerPosition = Networking.LocalPlayer.GetPosition();
+            float distance = Vector3.Distance(playerPosition, testHelpers.spawn2.position);
+            Assert.IsTrue(distance < 0.25f, "Player failed to respawn.");
+            
+            // Ensure OnPlayerRespawn event fired by checking variable set by this event
+            Assert.IsTrue(testHelpers.respawnWithIndexCube.GetProgramVariable<bool>(_programVariableRespawned));
+            
+            // Ensure OnPlayerRespawn fired by EventDispatcher
+            Assert.IsTrue(respawnEventFromDispatcherTriggered);
+            EventDispatcher.Unsubscribe<ClientSimOnPlayerRespawnEvent>(OnPlayerRespawn);
+        }
+
+        [UnityTest]
+        public IEnumerator RemotePlayer_RespawnedViaApi_WillRespawn()
+        {
+            yield return WaitForClientSimStartup();
+            
+            // Verify initial state of objects in the scene.
+            var testHelpers = Object.FindObjectOfType<ClientSimIssue3RespawnTestObjectReferences>();
+            Assert.IsNotNull(testHelpers, "Could not find Test helper reference.");
+            
+            // Listen for OnPlayerRespawn event through dispatcher
+            bool respawnEventFromDispatcherTriggered = false;
+            void OnPlayerRespawn(ClientSimOnPlayerRespawnEvent respawnEvent)
+            {
+                respawnEventFromDispatcherTriggered = true;
+            }
+            EventDispatcher.Subscribe<ClientSimOnPlayerRespawnEvent>(OnPlayerRespawn);
+
+            // Call Respawn static method on PlayerManager
+            var remotePlayer = Helper.SpawnRemotePlayer("RemotePlayer");
+            ClientSimPlayerManager.Respawn(remotePlayer);
+            
+            // Ensure OnPlayerRespawn fired by EventDispatcher
+            Assert.IsTrue(respawnEventFromDispatcherTriggered);
```

Comment:
If I'm reading this right, you are trying to respawn a remote player. Now that the api logs a warning for non local players, wouldn't this test fail as `respawnEventFromDispatcherTriggered` should still be false? It feels like a bug in ClientSim if a remote player can trigger the player respawned event. 

In some of the other tests, logs are verified when we expect the method to return early. It might be worth adding that here.
https://github.com/vrchat-community/ClientSim/blob/7b3a59c809730f8d47da31f3aeaa93e42e88786f/Tests/Runtime/IntegrationTests/ClientSimPlayerApiTests.cs#L453